File: Leetcode/2011_find_value_of_variable_after_operation/solution.go
package main

import "fmt"

func findValue(operations []string) int {
	x := 0;

	for _, op := range operations {
		
		if op == "++X" || op == "X++" {
			x += 1;
		} else if op == "--X" || op == "X--" {
			x -= 1;
		}
	} 

	return x;
}

func main() {
	operations := []string {"++X", "++X", "--X", "X++", "X++"};

	value := findValue(operations);

	fmt.Println(value)
}

File: Leetcode/2011_find_value_of_variable_after_operation/solution.py
"""
	link: https://leetcode.com/problems/final-value-of-variable-after-performing-operations
	tag: Array, String
"""


def findValue(operations):
    # x = 0
    # for op in operations:
    #     if op == "++X" or op == "X++":
    #         x += 1
    #     elif op == "--X" or op == "X--":
    #         x -= 1
    # return x
    return len(operations) - str(operations).count('-')


print(findValue(["X++", "--X", "--X", "X--", "X++", "X++"]))
